handler-ssl-ocsp/src/main/java/io/netty/handler/ssl/ocsp/OcspClient.java

```diff
@@ -0,0 +1,316 @@
+
+                    // Get OCSP URL from Certificate and query it.
+                    URL uri = new URL(parseOcspUrlFromCertificate(x509Certificate));
+
+                    // Find port
+                    int port = uri.getPort();
+                    if (port == -1) {
+                        port = uri.getDefaultPort();
+                    }
+
+                    // Configure path
+                    String path = uri.getPath();
+                    if (path.isEmpty()) {
+                        path = "/";
+                    } else {
+                        if (uri.getQuery() != null) {
+                            path = path + "?" + uri.getQuery();
+                        }
+                    }
+
+                    Promise<OCSPResp> ocspResponsePromise = query(eventLoop,
+                            Unpooled.wrappedBuffer(builder.build().getEncoded()), uri.getHost(), port, path
+                            , channelFactory);
+
+                    // Validate OCSP response
+                    ocspResponsePromise.addListener(new GenericFutureListener<Future<OCSPResp>>() {
+                        @Override
+                        public void operationComplete(Future<OCSPResp> future) throws Exception {
+                            // If Future was successful then we have received OCSP response
+                            // We will now validate it.
+                            if (future.isSuccess()) {
+                                BasicOCSPResp resp = (BasicOCSPResp) future.get().getResponseObject();
+                                validateResponse(responsePromise, resp, derNonce, issuer, validateResponseNonce);
+                            } else {
+                                responsePromise.tryFailure(future.cause());
+                            }
+                        }
+                    });
+
+                } catch (Exception ex) {
+                    responsePromise.tryFailure(ex);
+                }
+            }
+        });
+        return responsePromise;
+    }
+
+    /**
+     * Query the OCSP responder for certificate status using HTTP/1.1
+     *
+     * @param eventLoop   {@link EventLoop} for HTTP request execution
+     * @param ocspRequest {@link ByteBuf} containing OCSP request data
+     * @param host        OCSP responder hostname
+     * @param port        OCSP responder port
+     * @param path        OCSP responder path
+     * @return Returns {@link Promise} containing {@link OCSPResp}
+     */
+    private static Promise<OCSPResp> query(final EventLoop eventLoop, final ByteBuf ocspRequest,
+                                           final String host, int port, final String path,
+                                           final ConnectionChannelFactory channelFactory) {
+        final Promise<OCSPResp> responsePromise = eventLoop.newPromise();
+
+        try {
+            Bootstrap bootstrap = new Bootstrap()
+                    .group(new NioEventLoopGroup().next())
```

Comment:
We're already being passed in an `eventLoop`, plus the `ConnectionChannelFactory` can produce one, and produces `socketChannel`s that might not be NIO channels.

Reply:
Oops! ;/